addons/auth_signup/models/res_partner.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import random
+import werkzeug
+
+from datetime import datetime, timedelta
+from urlparse import urljoin
+
+from openerp import api, fields, models, _
+
+from openerp.tools.misc import DEFAULT_SERVER_DATETIME_FORMAT, ustr
+
+
+class SignupError(Exception):
+    pass
+
+def random_token():
+    # the token has an entropy of about 120 bits (6 bits/char * 20 chars)
+    chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'
+    return ''.join(random.SystemRandom().choice(chars) for i in xrange(20))
+
+def now(**kwargs):
+    dt = datetime.now() + timedelta(**kwargs)
+    return dt.strftime(DEFAULT_SERVER_DATETIME_FORMAT)
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.multi
+    def _get_signup_valid(self):
+        for partner in self:
+            partner.signup_valid = bool(partner.signup_token) and \
+            (not partner.signup_expiration or now() <= partner.signup_expiration)
+
+    @api.v7
+    def _get_signup_url_for_action(self, cr, uid, ids, action=None, view_type=None, menu_id=None, res_id=None, model=None, context=None):
```

Comment:
I don't see any need to define this function with api.v7 and api.v8, it should work fine api.multi, shouldn't it?